cpp/lib/include/StringUtil.h

```diff
@@ -854,6 +854,13 @@ inline std::string BackslashEscape(const std::string &chars_to_escape, const std
 std::string ExtractHead(std::string * const target, const std::string &delimiter_string = " ", std::string::size_type start = 0);
 
 
+// \note If the delimiter is the first character in "s", an empty entry will be the first entry in the
+//       returned vector.  If the delimiter is the last entry in "s", no entry will be created at the end
```

Comment:
Why on earth would we want a common library function with such an inconsistent non-configurable standard behaviour?

Reply:
Please pardon my french... i just suggest that it would be better to either trim both sides or trim none. If more flexibility is needed, there could be either some kind of enum like e.g. TrimBehaviour or the caller just strips leading/trailing delimiters as needed manually before calling the function.

Reply:
I agree that this is tricky and it is not obvious to me what is best here. BUT, the typical delimiter is a newline so that means that we're splitting text.  If newlines are used to terminate lines, which is the typical use case, then the behaviour as implemented would be correct.  Please consider the following example:

\n
fred\n
bob\n

This would return 3 entries, the first of which would be empty.

Reply:
Moved into email_watcher.cc and renamed to SplitIntoLines (also dropped the delimiter parameter).